### ISSUE TITLE
Contribution Guidelines (CONTRIBUTING file or similar)

### ISSUE THREAD

**Issue by gerryhernandez** [ocornut/imgui#1596]
I've been following a few pull requests and I definitely appreciate the attention to keeping the repository clean. I think we could lessen some back and fourth by writing down some cursory contribution guidelines. This could manifest as a GitHub-canonical `CONTRIBUTING` file, or we can even stuff it into the `imgui.cpp` file, where the rest of the documentation lives, including the "how can I help" section.

Squashing, rebasing, and merging strategy is a low hanging fruit, just to get started.

Would this benefit anyone else?

**Comment by ocornut** [comment]
Hello, I've been meaning to do something like that. For now I have added guidelines for issue creation:
https://github.com/ocornut/imgui/blob/master/.github/CONTRIBUTING.md

I also made it that creating a new issue will have:

![image](https://user-images.githubusercontent.com/8225057/36106988-127672ba-1019-11e8-9dab-118da29f7a71.png)

I think Github UI shows a notification in the Issue Tracker for first time contributor, as I get a more prominent link to that file for other projects I don't work on.

We can later extend this to add pull request guidelines etc. At least I have one link I can point to when talking to users. Let me know if you have suggestion on the current document!
